### ISSUE TITLE
Add support for a `params` section per template

### ISSUE THREAD

**Issue by lquerel** [open-telemetry/weaver#288]
It’s not uncommon to have the same template applied with a different JQ filter and a different name for the generated file. Having the ability to specify a set of parameters per template specification in the weaver.yaml configuration file would be helpful. Today, it’s possible to simulate that by adding a projection in the JQ filter to include this information, but it’s not obvious and not necessarily easy to understand.

```yaml
params:
  excluded_attributes: ["messaging.client_id"]

templates:
  - pattern: "attributes.ts"
    filter: semconv_attributes({"exclude_stability": ["experimental"]})
    params:
      file_name: "stable_attributes.ts"
    application_mode: single
  - pattern: "attributes.ts"
    filter: semconv_attributes({"exclude_stability": ["stable"]})
    params:
      file_name: "experimental_attributes.ts"
    application_mode: single
```

**Comment by lmolkova** [comment]
It'd be nice to support `file_name` with a jinja template like `file_name: "{{ctx.output}}/{{ctx.root_namespace}}_attributes.ts"` or `file_name: "{{ctx.stability}}_attributes.ts"`.

Ideal stolen from @jsuereth approach to file name in build-tools 

https://github.com/open-telemetry/build-tools/blob/751485681aadcf9f47b51a9b5a9f6d91a9fc7ec8/semantic-conventions/README.md?plain=1#L341

**Comment by lquerel** [comment]
Implemented in #294
